validator/src/journal/chain.rs

```diff
@@ -639,30 +642,12 @@ impl<TEP: ExecutionPlatform + Clone + 'static, PV: PermissionVerifier + Clone +
 
         match result.status {
             BlockStatus::Valid => {
-                // Keep Ref-C: The block has been validated so ownership of the ext. ref. is
-                // maintained. The consensus engine is responsible for rendering an opinion of
-                // either commit, fail, or ignore, at which time the ext. ref. will be accounted
-                // for (moved into chain head in case of commit, dropped otherwise)
-                self.consensus_notifier.notify_block_new(block);
+                self.consensus_notifier
+                    .notify_block_valid(&block.header_signature);
             }
             BlockStatus::Invalid => {
-                let mut state = self
-                    .state
-                    .write()
-                    .expect("No lock holder should have poisoned the lock");
-
-                // Drop Ref-C: The block has been found to be invalid, and we are no longer
-                // interested in it. The invalid result will be cached for a period.
-                if state
```

Comment:
Don't we still need to drop this ref?  We're done with the block.

Reply:
The ref will get dropped when consensus decides on the block (commit, ignore, or fail). Maybe consensus will always decide fail though, so we can drop it immediately? I don't have a strong opinion either way.

Reply:
Put this drop back and added a commit to lower the logging level of the fail/ignore block methods (no need to warn on idempotent fail/ignore).